### ISSUE TITLE
[BUG] Can't lint go 1.22 package

### ISSUE THREAD

**Issue by Crocmagnon** [jjti/go-spancheck#19]
**Describe the bug**
I'm trying to lint a module requiring go 1.22.
Log:
```console
$ spancheck ./...
2024/04/22 10:44:34 [WARN] invalid start span signature "". expected regex:telemetry-type
/tmp/test/main.go:1:1: package requires newer Go version go1.22
ctrlflow: analysis skipped due to errors in package
spancheck: failed prerequisites: ctrlflow@example.com/test
```

**To Reproduce**
`go.mod`:
```
module example.com/test

go 1.22.2
```

`main.go`:
```
package main

import "fmt"

func main() {
	fmt.Println("hello, world")
}
```

**Expected behavior**
The package should be linted properly and, in this case, not report any error.
Also there's an unexpected warning about the new start span signature.

**Additional context**
I installed `spancheck` with `go1.22.2`:
```console
$ go version
go version go1.22.2 darwin/arm64
$ go install github.com/jjti/go-spancheck/cmd/spancheck@latest
```

**Comment by jjti** [comment]
Aside that's unrelated to the primary go version at hand, that `WARN` log line is from a small bug, just fixed that: https://github.com/jjti/go-spancheck/pull/20

**Comment by jjti** [comment]
@Crocmagnon I'm unfortunately unable to reproduce, running below:

```bash
# josh @ M-FTQC4VW0R3 in ~/jjti/go-spancheck on git:jjti/fix-warn-start-span o [19:15:06] 
$ go install github.com/jjti/go-spancheck/cmd/spancheck@latest
go: downloading github.com/jjti/go-spancheck v0.6.1

# josh @ M-FTQC4VW0R3 in ~/Desktop/main-test on git:main x [19:20:53] 
$ cat go.mod 
module github.com/jjti/test

go 1.22.2

# josh @ M-FTQC4VW0R3 in ~/Desktop/main-test on git:main x [19:20:55] 
$ ls
go.mod  main.go

# josh @ M-FTQC4VW0R3 in ~/Desktop/main-test on git:main x [19:20:56] 
$ cat main.go 
package main

import "fmt"

func main() {
        fmt.Println("hello, world")
}

# josh @ M-FTQC4VW0R3 in ~/Desktop/main-test on git:main x [19:20:58] 
$ spancheck ./...
```

**Comment by Crocmagnon** [comment]
I tried cloning the repo and building it with my local toolchain (go 1.21), I'm getting the same error as reported.
I then updated the toolchain to go1.22.2 with `go get toolchain@latest` and after building I'm still getting the same error.
However, if I instead update the go.mod to go1.22.2 (`go get go@latest`) and build, I'm not getting any error.

Tested on master, commit 128847493f00fe3b61d0c748873969e1d8ab42d1

**Comment by jjti** [comment]
@Crocmagnon I'm sorry to say I could never repro this issue. But I just upgraded the go version to 1.22.1 (in line with testifylint). Please let me know if you see the issue still after upgrading: https://github.com/jjti/go-spancheck/commit/926dd069e0fc58b2ea6aea274b440aa584ebc5e0
